dotnet/src/SemanticKernel/Text/TextChunker.cs

```diff
@@ -87,14 +101,14 @@ private static List<string> InternalSplitTextParagraphs(List<string> lines, int
         // Split long lines first
         IEnumerable<string> truncatedLines = lines.SelectMany(line => longLinesSplitter(line, adjustedMaxTokensPerParagraph));
 
-        var paragraphs = BuildParagraph(truncatedLines, adjustedMaxTokensPerParagraph, longLinesSplitter);
+        var paragraphs = BuildParagraph(truncatedLines, adjustedMaxTokensPerParagraph, longLinesSplitter, tokenCounter);
         // distribute text more evenly in the last paragraphs when the last paragraph is too short.
         if (paragraphs.Count > 1)
         {
             var lastParagraph = paragraphs[paragraphs.Count - 1];
             var secondLastParagraph = paragraphs[paragraphs.Count - 2];
 
-            if (TokenCount(lastParagraph.Length) < adjustedMaxTokensPerParagraph / 4)
+            if (tokenCounter(lastParagraph) < adjustedMaxTokensPerParagraph / 4)
```

Comment:
I'm curious now if we were ever supposed to be doing the division by 4 here -- I can't recall the reasoning....

Reply:
It's a rough check to see if the last paragraph is too short (less than 1/4 of the maximum tokens per paragraph) and if so, validates and merges it with the second to last paragraph. Best to leave as is and potentially refactor once the TextChunker is enhanced with some more configuration options?